File: PLP_Java/src/Aula03_ListaDeExercicio/Q02_Car.java
package Aula03_ListaDeExercicio;

// Defining the Car class
public class Q02_Car {
    // Class attributes
    private String brand;
    private String model;
    private int year;
    private int speed; // Speed initialized to 0

    // Class constructor
    public Q02_Car(String brand, String model, int year) {
        this.brand = brand;
        this.model = model;
        this.year = year;
        this.speed = 0; // Initial speed set to 0
    }

    // Method to display car details
    public void displayDetails() {
        System.out.println("Brand: " + brand + ", Model: " + model + ", Year: " + year);
    }

    // Method to accelerate the car
    public void accelerate(int amount) {
        this.speed += amount;
    }

    // Method to brake the car
    public void brake(int amount) {
        this.speed -= amount;
        if (this.speed < 0) {
            this.speed = 0; // Ensure the speed doesn't go below 0
        }
    }

    // Method to display the current speed
    public void displaySpeed() {
        System.out.println("Current speed: " + this.speed + " km/h");
    }

    // Main method to test the class
    public static void main(String[] args) {
        // Instantiating a car object
        Q02_Car car = new Q02_Car("Toyota", "Corolla", 2020);

        // Using the methods
        car.displayDetails();
        car.accelerate(50);
        car.displaySpeed();
        car.brake(20);
        car.displaySpeed();
        car.brake(40);
        car.displaySpeed();
    }
}

File: PLP_python/Aula03_ListaDeExercicios(15)/Q02_Car.py
class Q02_Car:

    def __init__(self, brand, model, year):
        self.brand = brand
        self.model = model
        self.year = year
        self.speed = 0  # Speed initialized to 0

    def display_details(self):
        """
        Prints the details of the car.
        """
        print('Brand: {}, Model: {}, Year: {}'.format(self.brand, self.model, self.year))

    def accelerate(self, amount):
        """
        Increases the speed of the car by a given amount.
        """
        self.speed += amount

    def brake(self, amount):
        """
        Decreases the speed of the car by a given amount, ensuring it doesn't go below 0.
        """
        self.speed -= amount
        if self.speed < 0:
            self.speed = 0

    def display_speed(self):
        """
        Displays the current speed of the car.
        """
        print('Current speed: {} km/h'.format(self.speed))


# Instantiating a car object
car = Q02_Car("Toyota", "Corolla", 2020)

# Using the methods
car.display_details()
car.accelerate(50)
car.display_speed()
car.brake(20)
car.display_speed()
car.brake(40)
car.display_speed()
